app/controllers/organizations/adopter_fosterer/likes_controller.rb

```diff
@@ -0,0 +1,59 @@
+class Organizations::AdopterFosterer::LikesController < Organizations::BaseController
+  before_action :authenticate_user!
+  before_action :set_pet, only: [:create]
+  before_action :set_like, only: [:destroy]
+  layout "adopter_foster_dashboard", only: :index
+
+  def index
+    authorize! with: Organizations::LikePolicy
+
+    @likes = current_user.adopter_foster_account.likes
+    @pets = current_user.adopter_foster_account.liked_pets
+  end
+
+  def create
+    authorize! context: {pet: @pet}, with: Organizations::LikePolicy
+
+    @like = Like.new(adopter_foster_account_id: current_user.adopter_foster_account.id,
+      pet_id: like_params[:pet_id])
+
+    respond_to do |format|
+      if @like.save
+        format.html { redirect_to request.referrer, notice: t("organizations.adopter_fosterer.likes.create.success", name: @pet.name) }
+        format.turbo_stream { flash.now[:notice] = t("organizations.adopter_fosterer.likes.create.success", name: @pet.name) }
+      else
+        format.html { redirect_to request.referrer, alert: t("organizations.adopter_fosterer.likes.create.failed", name: @pet.name) }
+        format.turbo_stream { flash.now[:alert] = t("organizations.adopter_fosterer.likes.create.failed", name: @pet.name) }
+      end
+    end
+  end
+
+  def destroy
+    @pet = @like.pet
+    authorize! context: {pet: @pet}, with: Organizations::LikePolicy
```

Comment:
I have some suggested refactoring here for the policy.

We can remove this `authorize!` line from `#destroy` and move it to `#set_like` but reword it as such:
```ruby
  def set_like
    @like = Like.find(params[:id])
    authorize! @like
  end
```

For actions that affect existing records (such as destroy, update, show), this is the pattern that should usually be used. We fetch the existing record, and then we authorize via that record. Action Policy can infer the correct policy based on the record's class. We don't need the pet context in this case. We should be performing checks directly on the record itself.

We can add this private method to the `LikePolicy`.
```ruby
  def pet
    @pet || record.pet
  end
```
This method will fetch the passed pet context if it exists (so for actions that do not have existing records). However, if the context is not given, it will then check if the passed record responds to `#pet`.